provider/pkg/provider/deployment_schedules.go

```diff
@@ -0,0 +1,345 @@
+	}
+	return &pulumirpc.DiffResponse{
+		Changes:             changes,
+		Replaces:            replaces,
+		DetailedDiff:        detailedDiffs,
+		HasDetailedDiff:     true,
+		DeleteBeforeReplace: len(replaces) > 0,
+	}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Delete(req *pulumirpc.DeleteRequest) (*pbempty.Empty, error) {
+	output, err := st.ToPulumiServiceDeploymentScheduleOutput(req.GetProperties())
+	if err != nil {
+		return nil, err
+	}
+
+	err = st.client.DeleteSchedule(context.Background(), output.Input.Stack, output.ScheduleID)
+	if err != nil {
+		return nil, err
+	}
+	return &pbempty.Empty{}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Create(req *pulumirpc.CreateRequest) (*pulumirpc.CreateResponse, error) {
+	input, err := st.ToPulumiServiceDeploymentScheduleInput(req.GetProperties())
+	if err != nil {
+		return nil, err
+	}
+
+	scheduleReq := pulumiapi.CreateDeploymentScheduleRequest{
+		ScheduleCron: input.ScheduleCron,
+		ScheduleOnce: input.ScheduleOnce,
+		Request: pulumiapi.CreateDeploymentRequest{
+			PulumiOperation: input.PulumiOperation,
+		},
+	}
+	scheduleID, err := st.client.CreateDeploymentSchedule(context.Background(), input.Stack, scheduleReq)
+	if err != nil {
+		return nil, err
+	}
+
+	outputProperties, err := plugin.MarshalProperties(
+		input.ToOutputPropertyMap(*scheduleID),
+		plugin.MarshalOptions{
+			KeepUnknowns: true,
+			SkipNulls:    true,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.CreateResponse{
+		Id:         path.Join(input.Stack.OrgName, input.Stack.ProjectName, input.Stack.StackName, *scheduleID),
+		Properties: outputProperties,
+	}, nil
+}
+
+func (st *PulumiServiceDeploymentScheduleResource) Check(req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
+	inputMap, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{KeepUnknowns: true, SkipNulls: true})
+	if err != nil {
+		return nil, err
+	}
+
+	var failures []*pulumirpc.CheckFailure
+	for _, p := range []resource.PropertyKey{"organization", "project", "stack", "pulumiOperation"} {
+		if !inputMap[(p)].HasValue() {
+			failures = append(failures, &pulumirpc.CheckFailure{
+				Reason:   fmt.Sprintf("missing required property '%s'", p),
+				Property: string(p),
+			})
+		}
+	}
+
+	if (inputMap["scheduleCron"].HasValue() && inputMap["timestamp"].HasValue()) ||
+		(!inputMap["scheduleCron"].HasValue() && !inputMap["timestamp"].HasValue()) {
+		failures = append(failures, &pulumirpc.CheckFailure{
+			Reason:   "One of scheduleCron and timestamp must be specified but not both",
+			Property: "projectName",
+		})
+	}
+
+	if inputMap["timestamp"].HasValue() && inputMap["timestamp"].IsString() {
+		_, err := time.Parse(time.RFC3339, inputMap["timestamp"].StringValue())
+		if err != nil {
+			failures = append(failures, &pulumirpc.CheckFailure{
+				Reason:   fmt.Sprintf("timestamp failed to parse due to: %s", err),
+				Property: "projectName",
```

Comment:
Should this be timestamp

Reply:
I did call the external property timestamp to make distinction easier. It is the same way in TTL, but not in raw API. (so logging timestamp to align with the name)

Totally open to renaming the property to `scheduleOnce` if that'd be better, I just think similar names are confusing + in TTL has timestamp.

Reply:
Oh I just meant the Property looks like a copy paste error
```suggestion
				Property: "timestamp",
```

Reply:
Oh crap, I didn't even notice that lol. Yes, def copypaste error, fixing